src/main/java/com/google/devtools/build/lib/skyframe/CompletionFunction.java

```diff
@@ -170,8 +172,20 @@ public SkyValue compute(SkyKey skyKey, Environment env)
     ValueT value = valueAndArtifactsToBuild.first;
     ArtifactsToBuild artifactsToBuild = valueAndArtifactsToBuild.second;
 
+    // Ensure that coverage artifacts are built before a target is considered completed.
     ImmutableList<Artifact> allArtifacts = artifactsToBuild.getAllArtifacts().toList();
-    SkyframeLookupResult inputDeps = env.getValuesAndExceptions(Artifact.keys(allArtifacts));
+    InstrumentedFilesInfo instrumentedFilesInfo =
+        value.getConfiguredObject().get(InstrumentedFilesInfo.STARLARK_CONSTRUCTOR);
+    Iterable<SkyKey> keysToRequest;
+    if (value.getConfiguredObject() instanceof ConfiguredTargetValue && instrumentedFilesInfo != null) {
```

Comment:
So this was the bug. Should `s/ConfiguredTargetValue/ConfiguredTarget`. After that all the tests passed.

Reply:
I pushed a fix, thanks!